src/test/java/api/CirculationRulesAPITests.java

```diff
@@ -155,20 +157,120 @@ public void cannotUpdateCirculationRulesWithOverdueFinePolicyId() {
+      UUID.randomUUID());
 
-    Response response = circulationRulesFixture
-      .attemptUpdateCirculationRules(rule, "i");
+    Response response = circulationRulesFixture.attemptUpdateCirculationRules(rule);
 
     assertThat(response.getStatusCode(), is(422));
     assertThat(response.getJson().getString("message"),
       is("The policy i does not exist"));
   }
 
+  @Test
+  public void canUpdateCirculationRulesWithTwentyExistingLoanPolicies() {
+
+    Set<UUID> loanPolicyIds = getSetOfPolicyIds(20);
+    createLoanPolicies(loanPolicyIds);
+
+    loanPolicyIds.forEach(loanPolicyId -> {
+      String rule = circulationRulesFixture.soleFallbackPolicyRule(
+        loanPolicyId, requestPoliciesFixture.allowAllRequestPolicy().getId(),
+        noticePoliciesFixture.activeNotice().getId(),
+        overdueFinePoliciesFixture.facultyStandard().getId(),
+        lostItemFeePoliciesFixture.facultyStandard().getId());
+      Response response = circulationRulesFixture.attemptUpdateCirculationRules(rule);
+      assertThat(String.format(
+        "Failed to set circulation rules: %s", response.getBody()),
+        response.getStatusCode(), is(204));
+    });
+  }
+
+  @Test
+  public void canUpdateCirculationRulesWithTwentyExistingNoticePolicies() {
+
+    Set<UUID> noticePolicyIds = getSetOfPolicyIds(20);
+    createNoticePolicies(noticePolicyIds);
+
+    noticePolicyIds.forEach(noticePolicyId -> {
+      String rule = circulationRulesFixture.soleFallbackPolicyRule(
+        loanPoliciesFixture.canCirculateFixed().getId(),
+        requestPoliciesFixture.allowAllRequestPolicy().getId(),
+        noticePolicyId,
+        overdueFinePoliciesFixture.facultyStandard().getId(),
+        lostItemFeePoliciesFixture.facultyStandard().getId());
+      Response response = circulationRulesFixture.attemptUpdateCirculationRules(rule);
+      assertThat(String.format(
+        "Failed to set circulation rules: %s", response.getBody()),
+        response.getStatusCode(), is(204));
+    });
+  }
+
+  @Test
+  public void canUpdateCirculationRulesWithTwentyExistingRequestPolicies() {
+
+    Set<UUID> requestPolicyIds = getSetOfPolicyIds(20);
+    requestPolicyIds.forEach(requestPoliciesFixture::allowAllRequestPolicy);
+
+    requestPolicyIds.forEach(requestPolicyId -> {
+      UUID allowAllRequestPolicyId = requestPoliciesFixture.allowAllRequestPolicy(
+        requestPolicyId).getId();
+      String rule = circulationRulesFixture.soleFallbackPolicyRule(
+        loanPoliciesFixture.canCirculateFixed().getId(),
+        allowAllRequestPolicyId,
+        noticePoliciesFixture.activeNotice().getId(),
+        overdueFinePoliciesFixture.facultyStandard().getId(),
+        lostItemFeePoliciesFixture.facultyStandard().getId());
+      Response response = circulationRulesFixture.attemptUpdateCirculationRules(rule);
+      assertThat(String.format(
+        "Failed to set circulation rules: %s", response.getBody()),
+        response.getStatusCode(), is(204));
+    });
```

Comment:
Personally, I might space out these statements to make it easier to spot the distinctions between them.

Reply:
done